components/sup/src/event.rs

```diff
@@ -177,6 +179,22 @@ pub fn service_stopped(service: &Service) {
     }
 }
 
+/// Send an event at the start of a Service update.
+pub fn service_update_started(service: &Service, update: &PackageIdent) {
+    if stream_initialized() {
+        publish(ServiceUpdateStartedEvent { event_metadata:   None,
+                                            service_metadata: Some(service.to_service_metadata()),
+                                            update_origin:    update.origin.clone(),
+                                            update_name:      update.name.clone(),
+                                            update_version:   update.version
+                                                                    .clone()
+                                                                    .unwrap_or_default(),
+                                            update_release:   update.release
+                                                                    .clone()
+                                                                    .unwrap_or_default(), });
```

Comment:
I think the `update` `PackageIdent` will always be a fully-qualified one at this point, so these could be made into `expect` calls.

However, this just drives home my desire to create a formal `FullyQualifiedPackageIdent` variant, though, so we can be 100% assured that this is the case 😭

Reply:
i thought i'd seen that only origin and name were required, but i'll double-check this and make the change if appropriate.

Reply:
In our current `PackageIdent` type that is true. However, in practice, identifiers without versions and releases (i.e., non-fully-qualified identifiers) only exist at the periphery of the system; once they make it down into the core, they're usually resolved to a concrete, fully-qualified identifier.

(If we actually had a `FullyQualifiedPackageIdent` type, then we could have the compiler _prove_ that for us, but alas, such a thing does not exist yet.)